ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift

```diff
@@ -133,25 +133,42 @@ class RoomSummaryProvider: RoomSummaryProviderProtocol {
         
         MXLog.verbose("\(name): Received \(diffs.count) diffs, current room list \(rooms.compactMap { $0.id ?? "Empty" })")
         
-        rooms = diffs
-            .reduce(rooms) { currentItems, diff in
-                guard let collectionDiff = buildDiff(from: diff, on: currentItems) else {
-                    MXLog.error("\(name): Failed building CollectionDifference from \(diff)")
-                    return currentItems
-                }
-                
-                guard let updatedItems = currentItems.applying(collectionDiff) else {
-                    MXLog.error("\(name): Failed applying diff: \(collectionDiff)")
-                    return currentItems
-                }
-                
-                return updatedItems
+        var updatedItems = rooms
+        for diff in diffs {
+            let resetDiffChunkingThreshhold = 50
+            if case .reset(let values) = diff, values.count > resetDiffChunkingThreshhold {
+                // Special case resets in order to prevent large updates from blocking the UI
+                // Render the first resetDiffChunkingThreshhold as a reset and then append the rest to give the UI time to update
+                updatedItems = processDiff(RoomListEntriesUpdate.reset(values: Array(values[..<resetDiffChunkingThreshhold])), on: updatedItems)
```

Comment:
Could we used the inferred type so there's less to visually parse when reading these manual diffs?
```suggestion
                updatedItems = processDiff(.reset(values: Array(values[..<resetDiffChunkingThreshhold])), on: updatedItems)
```